csrc/ops/composite.cpp

```diff
@@ -331,6 +335,31 @@ TensorView* matmul(TensorView* tv_a, TensorView* tv_b) {
       " and ",
       tv_b->dtype());
 
+  // Check for K=1 i.e. reduction of broadcast. In these cases we don't need a
+  // matmul so we translate it to a multiplication+cast
```

Comment:
Can we have a comment at the `matmul` definition or in the header enumerating the different cases and how they are handled for an easier summary.